scss/_patterns_icons.scss

```diff
@@ -89,6 +89,10 @@
       vertical-align: 0;
     }
   }
+
+  p > [class*='p-icon'] {
+    vertical-align: middle;
+  }
```

Comment:
This is using a child selector, while the selectors above it for heading levels are using descendant selectors.

Is there a reason this should only apply to direct child icons?

Reply:
Could this be done as default inside `%icon`?

https://github.com/canonical/vanilla-framework/blob/510f79dcac1b426f145f1324c214dc2f9f03993e/scss/_base_placeholders.scss#L228

Obviously this would affect all icons, but maybe it would be OK? (we can run it through Percy and check).
Current value is quite complicated calculation based on multiple variables (to the point that it's not clear what is actually being calculated there).

Reply:
I checked my suggestion and it would result in too many changes visually. Not worth it.